src/pydap/net.py

```diff
@@ -3,15 +3,18 @@
 
 from six.moves.urllib.parse import urlsplit, urlunsplit
 
-def GET(url, application=None):
+
+def GET(url, application=None, session=None):
```

Comment:
Passing the session object through is possibly a little verbose, but I do like it for following [the zen of Python](https://github.com/python/peps/blob/master/pep-0020.txt#L28) and it's far preferable to the mokeypatching solution (shudder).